### PR TITLE
Add Key Provisioning logic for requesting group keys from Hostlib

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR adds the Orchestrator logic for requesting the Hostlib with:
- Key Provisioning role request (whether it's a Leader or a Dependant)
- Requesting group keys for a Dependant role

Ref https://github.com/project-oak/oak/issues/4442

### CODE REVIEW

File: oak_containers_launcher/src/server.rs [hunk 1]
```diff
@@ -189,6 +195,27 @@ impl Launcher for LauncherServerImplementation {
...
+            role: KeyProvisioningRole::Leader.into(),
+        }))
+    }
+
+    async fn get_group_keys(
+        &self,
+        _request: Request<()>,
+    ) -> Result<Response<GetGroupKeysResponse>, tonic::Status> {
+        // TODO(#4442): Implement sending group keys to the orchestrator.
+        Ok(tonic::Response::new(GetGroupKeysResponse::default()))
```
Comment (tiziano88) [inline oak_containers_launcher/src/server.rs:215]: Why does this return Ok? Shouldn't it return Err?

File: oak_containers_orchestrator/src/main.rs [hunk 2]
```diff
@@ -94,6 +96,24 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         tokio::fs::create_dir_all(path).await?;
     }
 
+    let key_provisioning_role = launcher_client
+        .get_key_provisioning_role()
+        .await
+        .map_err(|error| anyhow!("couldn't get key provisioning role: {:?}", error))?;
+    let key_store = Arc::new(match key_provisioning_role {
+        KeyProvisioningRole::Unspecified => Err(anyhow!("unspecified key provisioning role"))?,
```
Comment (tiziano88) [inline oak_containers_orchestrator/src/main.rs:104]: I think `Err(anyhow!(...))?` can be simplified as `anyhow::bail!(...)`?

**Review by tiziano88 — APPROVED** [review]

File: oak_containers_orchestrator/src/main.rs (hunk 2 above)
Comment (ipetr0v) [reply oak_containers_orchestrator/src/main.rs]: Done

File: oak_containers_launcher/src/server.rs (hunk 1 above)
Comment (ipetr0v) [reply oak_containers_launcher/src/server.rs]: Done

**Review by pmcgrath17 — APPROVED** [review]
Should we consider changing the name of the ```oak_containers_launcher``` directory to ```oak_containers_hostlib```?

**Pull request merged**